src/modules/lesson/controllers/lesson.controller.ts

```diff
@@ -34,88 +36,121 @@ import { LessonEntity } from '../entities/lesson.entity';
+  }
 
-    updatedLesson.hashtags = updatedLessonHashtags;
+  @ApiOperation({ summary: '과제 삭제' })
+  @ApiSuccessResponse(HttpStatus.OK, {
+    lesson: OmitType(LessonEntity, ['hashtags']),
+  })
+  @ApiFailureResponse(HttpStatus.FORBIDDEN, 'You do not have access to ~')
+  @ApiFailureResponse(HttpStatus.NOT_FOUND, "~ doesn't exist id in ~")
+  @BearerAuth(JwtAuthGuard)
+  @Delete(':id')
+  async deleteLesson(
+    @Param()
+    @SetModelNameToParam(ModelName.Lesson)
+    param: IdRequestParamDto,
+    @UserLogin('id') memberId: number,
+  ): Promise<{ lesson: Omit<LessonEntity, 'hashtag'> }> {
+    await this.prismaHelper.validateOwnerOrFail(ModelName.Lesson, {
+      id: param.id,
+      memberId,
+    });
+    const deletedLesson = await this.lessonService.deleteLesson(param.id);
 
-    return plainToInstance(LessonEntity, updatedLesson);
+    return { lesson: deletedLesson };
   }
 
   @ApiOperation({ summary: '과제 상세 조회' })
-  @ApiOkResponse({ type: ReadOneLessonDto })
-  @CustomApiResponse(
-    HttpStatus.NOT_FOUND,
-    "(과제 번호) doesn't exist id in lesson",
-  )
+  @ApiSuccessResponse(HttpStatus.OK, { lesson: ReadOneLessonDto })
+  @ApiFailureResponse(HttpStatus.NOT_FOUND, "~ doesn't exist id in ~")
   @BearerAuth(OptionalJwtAuthGuard)
   @Get(':id')
   async readOneLesson(
-    @Param()
-    @SetModelNameToParam(ModelName.Lesson)
-    param: IdRequestParamDto,
+    @Param() @SetModelNameToParam(ModelName.Lesson) param: IdRequestParamDto,
     @UserLogin() member: Member,
-  ): Promise<ReadOneLessonDto> {
-    const lesson = await this.lessonService.readOneLesson(param.id, member.id);
+  ): Promise<{ lesson: ReadOneLessonDto }> {
+    const readOneLesson = await this.lessonService.readOneLesson(
+      param.id,
+      member.id,
+    );
+    const lesson = plainToInstance(ReadOneLessonDto, readOneLesson);
 
-    return plainToInstance(ReadOneLessonDto, lesson);
+    return { lesson };
   }
 
   @ApiOperation({ summary: '과제 상세 조회의 유사과제' })
   @ApiOkResponse({ type: ReadSimilarLessonDto })
-  @CustomApiResponse(
-    HttpStatus.NOT_FOUND,
-    "(과제 번호) doesn't exist id in Lesson",
-  )
+  @ApiFailureResponse(HttpStatus.NOT_FOUND, "~ doesn't exist id in ~")
   @BearerAuth(OptionalJwtAuthGuard)
   @Get(':id/similarity')
   async readSimilarLesson(
-    @Param() @SetModelNameToParam(ModelName.Lesson) param: IdRequestParamDto,
+    @Param()
+    @SetModelNameToParam(ModelName.Lesson)
+    param: IdRequestParamDto,
     @Query() query: SimilarLessonQueryDto,
     @UserLogin() member: Member,
   ): Promise<ReadSimilarLessonDto> {
-    const lessons = await this.lessonService.readSimilarLesson(
+    const readSimilarLessons = await this.lessonService.readSimilarLesson(
       param.id,
       member.id,
       query,
     );
-
-    return plainToInstance(ReadSimilarLessonDto, {
-      lessons,
+    const lessons = plainToInstance(ReadSimilarLessonDto, {
+      lessons: readSimilarLessons,
     });
+
+    return lessons;
```

Comment:
```suggestion
    return plainToInstance(ReadSimilarLessonDto, {
      lessons: readSimilarLessons,
    });
```
제 에디터 설정에서는 이렇게 고치는거 어떠냐고 물어보네요

Reply:
가만 생각해 보니 의미 없는 변수 할당이네요